bot/seasons/evergreen/bookmark.py

```diff
@@ -17,17 +20,27 @@ def __init__(self, bot: commands.Bot):
 
     @commands.command(name="bookmark", aliases=("bm", "pin"))
     async def bookmark(
-        self,
-        ctx: commands.Context,
-        target_message: discord.Message,
-        *,
-        title: str = "Bookmark"
+            self,
+            ctx: commands.Context,
+            target: Union[discord.Message, discord.User],
+            *,
+            title: str = "Bookmark"
     ) -> None:
-        """Send the author a link to `target_message` via DMs."""
+        """
+        Send the author a link to `target_message` via DMs.
+
+        A user mention can also be passed to get the user's last message in channel where the command is invoked.
+        """
+        if type(target) == discord.User:
+            async for message in ctx.channel.history(limit=20):  # Takes last 2 messages from the channel.
+                if message.author == target:
+                    target = message
+                    break
+
         # Prevent users from bookmarking a message in a channel they don't have access to
-        permissions = ctx.author.permissions_in(target_message.channel)
+        permissions = ctx.author.permissions_in(target.channel)
```

Comment:
This will fail when doing a `.bm @seasonalbot` raising an error `AttributeError: 'ClientUser' object has no attribute 'channel'` - You will need to check if target is a bot as well.

Reply:
Added a check. so resolving this convo

Reply:
This has not been solved yet, when I do `.bm @seasonalbot hi` I got this traceback
```py
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/discord/ext/commands/core.py", line 83, in wrapped
    ret = await coro(*args, **kwargs)
  File "/bot/bot/seasons/evergreen/bookmark.py", line 46, in bookmark
    permissions = ctx.author.permissions_in(target.channel)
AttributeError: 'ClientUser' object has no attribute 'channel'
```